specs/ReleaseChannelPreferenceSet.md

```diff
@@ -0,0 +1,325 @@
+  COREWEBVIEW2_RUNTIME_CHANNEL_CANARY,
+} COREWEBVIEW2_RUNTIME_CHANNEL;
+cpp_quote("DEFINE_ENUM_FLAG_OPERATORS(COREWEBVIEW2_RUNTIME_CHANNEL)")
+
+/// Additional options used to create the WebView2 Environment that support
+/// specifying the `RuntimeChannelPreferenceSet` and `ReverseChannelSearchOrder`.
+[uuid(47ac856e-a726-4e04-b36b-f58dafb39e38), object, pointer_default(unique)]
+interface ICoreWebView2EnvironmentOptions6 : ICoreWebView2EnvironmentOptions5 {
+
+  /// Gets the `RuntimeChannelPreferenceSet`.
+  HRESULT GetRuntimeChannelPreferenceSet(
+      [out] COREWEBVIEW2_RUNTIME_CHANNEL* channels);
+
+  /// Sets the `RuntimeChannelPreferenceSet`, which is a mask of one or more
+  /// `COREWEBVIEW2_RUNTIME_CHANNEL`s. OR operation(s) can be applied to multiple
+  /// `COREWEBVIEW2_RUNTIME_CHANNEL`s to create a mask. The default value is a
+  /// a mask of all the channels. By default, the WebView2 loader
+  /// searches for channels from most to least stable, using the first channel
+  /// found on the device. When a `RuntimeChannelPreferenceSet` is provided, the
+  /// WebView2 loader will only search for the channels specified in the set.
+  /// Set `ReverseChannelSearchOrder` to `TRUE` to reverse the search order so
+  /// that the loader searches for least stable build first. See
+  /// `COREWEBVIEW2_RUNTIME_CHANNEL` for descriptions of each channel.
+  /// `CreateCoreWebView2EnvironmentWithOptions` fails with
+  /// `HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND)` if the loader is unable to find
+  /// any channel from the `RuntimeChannelPreferenceSet` installed on the device.
+  /// Use `GetAvailableCoreWebView2BrowserVersionStringWithOptions` on
+  /// `ICoreWebView2Environment` to verify which channel is used when this option
+  /// is set.
+  ///
+  /// Examples:
+  /// |   RuntimeChannelPreferenceSet   |   Default Channel Search Order   |   Reverse Channel Search Order   |
+  /// | --- | --- | --- |
+  /// |COREWEBVIEW2_RUNTIME_CHANNEL_BETA \| COREWEBVIEW2_RUNTIME_CHANNEL_STABLE| WebView2 Runtime -> Beta | Beta -> WebView2 Runtime|
+  /// |COREWEBVIEW2_RUNTIME_CHANNEL_CANARY \| COREWEBVIEW2_RUNTIME_CHANNEL_DEV \| COREWEBVIEW2_RUNTIME_CHANNEL_BETA \| COREWEBVIEW2_RUNTIME_CHANNEL_STABLE| WebView2 Runtime -> Beta -> Dev -> Canary | Canary -> Dev -> Beta -> WebView2 Runtime |
+  /// |COREWEBVIEW2_RUNTIME_CHANNEL_CANARY| Canary | Canary |
+  /// |COREWEBVIEW2_RUNTIME_CHANNEL_BETA \| COREWEBVIEW2_RUNTIME_CHANNEL_CANARY \| COREWEBVIEW2_RUNTIME_CHANNEL_STABLE | WebView2 Runtime -> Beta -> Canary | Canary -> Beta -> WebView2 Runtime |
+  ///
+  /// If both a `BrowserExecutableFolder` and a `RuntimeChannelPreferenceSet` are
+  /// provided, the `BrowserExecutableFolder` takes precedence. The
+  /// `RuntimeChannelPreferenceSet` can be overridden by the corresponding
+  /// registry override `WEBVIEW2_RUNTIME_CHANNEL_PREFERENCE_SET` or the
+  /// environment variable `RuntimeChannelPreferenceSet`. Set the value to a
```

Comment:
It's weird that the registry overrides the app setting. I would have expected the app setting to override the registry, so that app developers can force a specific channel and not worry about running on a machine that has been reconfigured to prefer some other channel.

Reply:
The scenario is for IT and dev so it definitely needs to override the app's settings. Note in the docs that because of this override you may get a different runtime than what you specified.